elide-core/src/main/java/com/yahoo/elide/security/PermissionManager.java

```diff
@@ -0,0 +1,500 @@
+        CheckMode classCheckMode = null;
+        final ArrayList<CheckMode> fieldCheckModes = new ArrayList<>();
+
+        A annotation = dictionary.getAnnotation(resource, annotationClass);
+        if (annotation != null) {
+            ExtractedChecks extracted = extractChecks(annotationClass, annotation);
+            classOpChecks = extracted.getOperationChecks();
+            classComChecks = extracted.getCommitChecks();
+            classCheckMode = extracted.getCheckMode();
+        }
+
+        // Get all fields
+        Class<?> entityClass = resource.getResourceClass();
+        List<String> attributes = dictionary.getAttributes(entityClass);
+        List<String> relationships = dictionary.getRelationships(entityClass);
+        List<String> fields = (attributes != null) ? new ArrayList<>(dictionary.getAttributes(entityClass))
+                                                   : new ArrayList<>();
+        if (relationships != null) {
+            fields.addAll(relationships);
+        }
+
+        for (String field : fields) {
+            annotation = dictionary.getAttributeOrRelationAnnotation(entityClass, annotationClass, field);
+            if (annotation == null) {
+                continue;
+            }
+            ExtractedChecks extracted = extractChecks(annotationClass, annotation);
+            fieldOpChecks.add(extracted.getOperationChecks());
+            fieldComChecks.add(extracted.getCommitChecks());
+            fieldCheckModes.add(extracted.getCheckMode());
+        }
+
+        boolean hasDeferredChecks = (!isEmptyCheckArray(classComChecks) || !isEmptyListOfChecks(fieldComChecks))
+                && (annotationClass.equals(UpdatePermission.class) || annotationClass.equals(CreatePermission.class));
+
+        // Run operation checks
+        final AnyField strategy = new AnyField();
+        fieldAwareExecute(classOpChecks, fieldOpChecks, classCheckMode, fieldCheckModes, resource,
+                changeSpec, hasDeferredChecks, strategy);
+
+        // Need these to be final so we can capture within lambda
+        final Class<? extends Check>[] captureClassComChecks = classComChecks;
+        final CheckMode captureClassCheckMode = classCheckMode;
+
+        // If that succeeds, queue up the commit checks
+        if (hasDeferredChecks) {
+            commitChecks.add(() -> {
+                fieldAwareExecute(captureClassComChecks, fieldComChecks, captureClassCheckMode,
+                        fieldCheckModes, resource, changeSpec, false, strategy);
+                return null;
+            });
+        }
+    }
+
+    /**
+     * Check for permissions on a specific field.
+     *
+     * @param resource resource
+     * @param changeSpec changepsec
+     * @param annotationClass annotation class
+     * @param field field to check
+     * @param <A> type parameter
+     */
+    public <A extends Annotation> void checkFieldAwarePermissions(PersistentResource<?> resource,
```

Comment:
I feel like we can keep these two entry functions, but they can both just call a helper function that takes a strategy and a set of fields (could be a single field) where permissions need to be checked.  That function will do more or less the same setup (extract the checks for the entity, extract the checks for field(s), call fieldAwareExecute with the strategy and the associated extracted checks.